### ISSUE TITLE
Role template does not cover permissions to create services

### ISSUE THREAD

**Issue by hantonio** [timescale/timescaledb-kubernetes#405]
**Describe the bug**
Installing the latest timescaledb-single helm chart (0.14.0) with `helm install timescaledb ./timescaledb-single`, I repeatedly found the same error message related to lack of permissions for serviceaccount timescaledb.
Indeed, adding the following lines to `timescaledb-single/templates/role-timescaledb.yaml` template solved the problem.
```
- apiGroups: [""]
  resources: ["services"]
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
  # delete is required only for 'patronictl remove'
  - delete
```
I'm just an end-user, but I hope this is a suggestion for whoever maintains the chart.

**To Reproduce**
Install the latest timescaledb-single helm chart (0.14.0) with `helm install timescaledb ./timescaledb-single`

**Expected behavior**
No errors related to service creation permissions that would prevent timescaledb installation.

**Deployment**
- What is in your `values.yaml`? 
Standard values.yaml from the repo.
- What version of the Chart are you using?
0.14.0
- What is your Kubernetes Environment (for exampe: GKE, EKS, minikube, microk8s)
RKE2 (Rancker Kubernetes Engine 2)

**Deployment**
Please share some details of what is in your Kubernetes environment, for example:

After the change in the role manifest, everything seems normal.

```
kubectl get all,secret,configmap,endpoints,pvc -L role -l release=$RELEASE
NAME                READY   STATUS    RESTARTS   AGE    ROLE
pod/timescaledb-0   1/1     Running   0          11m    master
pod/timescaledb-1   1/1     Running   0          10m    replica
pod/timescaledb-2   1/1     Running   0          8m6s   replica

NAME                          TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE   ROLE
service/timescaledb           LoadBalancer   10.43.225.163   <pending>     5432:30465/TCP   11m   master
service/timescaledb-config    ClusterIP      None            <none>        8008/TCP         11m
service/timescaledb-replica   ClusterIP      10.43.19.87     <none>        5432/TCP         11m   replica

NAME                           READY   AGE   ROLE
statefulset.apps/timescaledb   3/3     11m

NAME                               DATA   AGE   ROLE
configmap/timescaledb-patroni      1      11m
configmap/timescaledb-pgbackrest   1      11m
configmap/timescaledb-scripts      8      11m

NAME                            ENDPOINTS                                            AGE   ROLE
endpoints/timescaledb           10.42.5.151:5432                                     10m
endpoints/timescaledb-config    10.42.3.153:8008,10.42.4.190:8008,10.42.5.151:8008   11m
endpoints/timescaledb-replica   10.42.3.153:5432,10.42.4.190:5432                    11m   replica

NAME                                                 STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE    ROLE
persistentvolumeclaim/storage-volume-timescaledb-0   Bound    pvc-902b2f72-8784-4422-9d71-d75e4373a2d5   2Gi        RWO            longhorn       11m
persistentvolumeclaim/storage-volume-timescaledb-1   Bound    pvc-139ac2e8-b237-4761-9f1e-d20d9df154d9   2Gi        RWO            longhorn       10m
persistentvolumeclaim/storage-volume-timescaledb-2   Bound    pvc-2d41f2e1-268f-4207-9894-4a940691bbc8   2Gi        RWO            longhorn       8m7s
persistentvolumeclaim/wal-volume-timescaledb-0       Bound    pvc-b4ba03f8-48e5-4bfe-9ea5-47a459e1b705   1Gi        RWO            longhorn       11m
persistentvolumeclaim/wal-volume-timescaledb-1       Bound    pvc-e9aa4bb2-0d2c-4fdc-98e3-e17a1b3aaf31   1Gi        RWO            longhorn       10m
persistentvolumeclaim/wal-volume-timescaledb-2       Bound    pvc-8aaba288-0334-45c4-9861-d86e821b8655   1Gi        RWO            longhorn       8m7s
```

**Logs**

```
2022-07-26 20:16:51,007 ERROR: create_config_service failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/patroni/dcs/kubernetes.py", line 890, in _create_config_service
    if not self._api.create_namespaced_service(self._namespace, body):
  File "/usr/lib/python3/dist-packages/patroni/dcs/kubernetes.py", line 468, in wrapper
    return getattr(self._core_v1_api, func)(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/patroni/dcs/kubernetes.py", line 404, in wrapper
    return self._api_client.call_api(method, path, headers, body, **kwargs)
  File "/usr/lib/python3/dist-packages/patroni/dcs/kubernetes.py", line 373, in call_api
    return self._handle_server_response(response, _preload_content)
  File "/usr/lib/python3/dist-packages/patroni/dcs/kubernetes.py", line 203, in _handle_server_response
    raise k8s_client.rest.ApiException(http_resp=response)
patroni.dcs.kubernetes.K8sClient.rest.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Kubernetes-Pf-Flowschema-Uid': '071a76d2-7a31-4767-a914-6427db79e62d', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'be8aeb3b-60e0-42be-99e8-144696334148', 'Date': 'Tue, 26 Jul 2022 20:16:51 GMT', 'Content-Length': '300'})
HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services is forbidden: User \\"system:serviceaccount:opennms:timescaledb\\" cannot create resource \\"services\\" in API group \\"\\" in the namespace \\"opennms\\"","reason":"Forbidden","details":{"kind":"services"},"code":403}\n'
```

**Additional context**
Nothing to add here.

**Comment by nhudson** [comment]
From #402 I believe the issue stems from commit https://github.com/timescale/timescaledb-kubernetes/commit/b6265f3b45503a454f3e4da316acc166201ff96d

**Comment by riannoor** [comment]
> From #402 I believe the issue stems from commit [b6265f3](https://github.com/timescale/timescaledb-kubernetes/commit/b6265f3b45503a454f3e4da316acc166201ff96d)

Seems to be, I faced the same issue yesterday then I added this rule that @hantonio suggested to fix it:
```
- apiGroups: [""]
  resources: ["services"]
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
  # delete is required only for 'patronictl remove'
  - delete
 ```

Then just now I reduced the rule similar to the one removed by commit https://github.com/timescale/timescaledb-kubernetes/commit/b6265f3b45503a454f3e4da316acc166201ff96d and it did the trick

```
- apiGroups: [""]
   resources: ["services"]
   verbs:
   - create
```

**Comment by paulfantom** [comment]
"Fix" described here is just a workaround and it is not fixing the root cause which is in patroni itself - https://github.com/zalando/patroni/issues/1132.

Patroni [code docs clearly state that this RBAC permission is **not** needed](https://github.com/zalando/patroni/blob/0b1b1e3b542714b47634391f7dbb00016d885bff/kubernetes/patroni_k8s.yaml#L190-L193) when Service object is managed by external entities and our helm chart is obeying this instruction.

If we apply the workaround here, stack uninstallation would be broken as patroni would be racing with helm to recreate the Service object. We opted to remove the RBAC permission as it shouldn't affect patroni operations (apart from the spam log message) and we intend to fix the root cause in the future. If you would like to see this fix earlier, feel free to send a PR towards patroni repository :)

**Comment by jleni** [comment]
I would add that this has been fixed in Patroni recently: 
https://github.com/zalando/patroni/pull/2390

**Comment by fhisamoto** [comment]
I'm still seeing the error. I'm installing timescale-db using the tobs operator helm chart latest release: `16.1.0`.

**Comment by nhudson** [comment]
> I'm still seeing the error. I'm installing timescale-db using the tobs operator helm chart latest release: `16.1.0`.

@fhisamoto the change to patroni has not been merged into the Timescale container image yet.  Should be soon.

**Comment by nhudson** [comment]
> For info: this is not fixed in `timescaledb-single` v.0.19.0 (latest as of 1 hour ago)

Correct.  This fix is not helm chart related, but yet an issue with Patroni in the Timescaledb container image.  Until that is updated with the fix from upstream Patroni this will not be updated in the helm chart.

**Comment by WillBeebe** [comment]
If you're waiting on the fix you can apply the role and binding yourself as a workaround.

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  namespace: default
  name: timescale-patch
rules:
- apiGroups: [""] # "" indicates the core API group
  resources: ["services"]
  verbs: ["create"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: timescale-patch-binding
  namespace: default
subjects:
- kind: ServiceAccount
  name: the-timescaledb-release-name
  namespace: default
roleRef:
  kind: Role 
  name: timescale-patch
  apiGroup: rbac.authorization.k8s.io
```

**Comment by dungnguyen1254** [comment]
Still facing the issue with version 0.26.1

**Comment by vietnguyen1254** [comment]
Still facing the issue with version 0.26.1

**Comment by nhudson** [comment]
The default timescaledb image doesn't have the correct changes from Patroni in it. You will need to use this container image for now, until the default image gets updated with the latest release of Patroni. 

`timescale/timescaledb-ha:pg14.6-ts2.8.1-patroni-static-primary-p3`

This container image has the Patroni fix patched into it.

**Comment by vietnguyen1254** [comment]
> The default timescaledb image doesn't have the correct changes from Patroni in it. You will need to use this container image for now, until the default image gets updated with the latest release of Patroni.
> 
> `timescale/timescaledb-ha:pg14.6-ts2.8.1-patroni-static-primary-p3`
> 
> This container image has the Patroni fix patched into it.

Thank you so much! It works like a charm!

**Comment by throrin19** [comment]
Any news about that ? It's fixed in the last version of timescaledb-single or not ?

**Comment by waldner** [comment]
Seeing this with 0.27.4 too.

**Comment by mogopz** [comment]
@paulfantom Can we please re-open this? It's still an issue on the latest Postgres v13 image (`pg13.10-ts2.9-latest`).

**Comment by throrin19** [comment]
@mogopz I open an issue with the same problem here : #554

**Comment by aman-github-ns** [comment]
I am still facing the issue. I am using this version
    repository: timescale/timescaledb-ha
    tag: pg14.6-ts2.9.1-p1

**Comment by pixelfields** [comment]
> I am still facing the issue. I am using this version repository: timescale/timescaledb-ha tag: pg14.6-ts2.9.1-p1

The same for me, the fresh install, the image is `timescale/timescaledb-ha:pg14.6-ts2.9.1-p1`

**Comment by IvaskevychYuriy** [comment]
We've faced the same issue with
- chart version `0.33.1`
- Docker image `pg14.6-ts2.9.1-p1`

As pointed out by others - this issue was in Patroni release and the fix was already available in [`2.1.5`](https://patroni.readthedocs.io/en/latest/releases.html#version-2-1-5)

But the Patroni is shipped with the `timescale/timescaledb-ha` image so if you check it out:
`kubectl exec -it timescaledb-0 -- sh`
`patronictl version`
It reports `2.1.4`

So we decided to use existing chart version `0.33.1` but migrate to a newer TSDB Docker image - `pg15.4-ts2.12.2` (it was the latest one available at the moment of writing this) by overriding `values.yaml` with:
```yaml
image:
  tag: pg15.4-ts2.12.2
```

This one is shipped with Patroni `3.1.0` (if you check it inside the timescale-db pod) and fixes the issue.

**NOTE**: We did not migrate any data as this was a local dev environment so we just deleted existing volumes before upgrading to image with newer Postgres\Timescale
